### PR TITLE
Dev15 interactive window toolbar support

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
I think I'll merge when we have build for external users but in time for Dev15 injection.

### CODE REVIEW

File: src/Mocks/Editor/TextSelectionMock.cs [hunk 1]
```diff
@@ -9,46 +9,36 @@
...
             _range = range;
         }
 
         public bool ActivationTracksFocus { get; set; }
 
-        public VirtualSnapshotPoint ActivePoint
-        {
-            get
-            {
+        public VirtualSnapshotPoint ActivePoint {
```
Comment (AlexanderSher) [inline src/Mocks/Editor/TextSelectionMock.cs]: If you're changing formatting here, you can reduce it to `public VirtualSnapshotPoint ActivePoint => AnchorPoint;`

File: src/Mocks/Editor/project.15.0.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
 ﻿{
   "dependencies": {
     "MicroBuild.Core": "0.2.0",
-    "Microsoft.VisualStudio.CoreUtility": "15.0.25824-RC",
+    "Microsoft.VisualStudio.Text.UI.Wpf": "15.0.25901-RC",
```
Comment (AlexanderSher) [inline src/Mocks/Editor/project.15.0.json]: Duplicated `Microsoft.VisualStudio.Text.UI.Wpf`

File: src/Mocks/VisualStudio/project.15.0.json [hunk 3]
```diff
@@ -1,21 +1,21 @@
 ﻿{
   "dependencies": {
     "MicroBuild.Core": "0.2.0",
-    "Microsoft.VisualStudio.CoreUtility": "15.0.25824-RC",
+    "Microsoft.VisualStudio.ComponentModelHost": "15.0.25824-RC",
```
Comment (AlexanderSher) [inline src/Mocks/VisualStudio/project.15.0.json]: Condition above includes `Microsoft.VisualStudio.ComponentModelHost` only for VS14. Do w still need it in `project.json`?

File: src/Mocks/VisualStudio/project.15.0.json [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-    "Microsoft.VisualStudio.Text.UI.Wpf": "15.0.25824-RC",
+    "Microsoft.VisualStudio.Text.Data": "15.0.25901-RC",
+    "Microsoft.VisualStudio.Text.Logic": "15.0.25901-RC",
+    "Microsoft.VisualStudio.Text.UI": "15.0.25901-RC",
+    "Microsoft.VisualStudio.Text.UI.Wpf": "15.0.25901-RC",
     "Microsoft.VisualStudio.TextManager.Interop.11.0": "11.0.61030",
     "Microsoft.VisualStudio.TextManager.Interop.12.0": "12.0.30110",
     "NSubstitute": "1.10.0",
-    "System.Collections.Immutable": "1.3.0",    
+    "System.Collections.Immutable": "1.2.0",
```
Comment (AlexanderSher) [inline src/Mocks/VisualStudio/project.15.0.json]: Downgraded `System.Collections.Immutable`?

File: src/Package/Impl/Repl/Commands/History/NextHistoryReplCommand.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.VisualStudio.R.Package.Commands;
+
+namespace Microsoft.VisualStudio.R.Package.Repl.Commands {
+    internal sealed class NextHistoryReplCommand : ReplCommandBase {
+        public NextHistoryReplCommand(IRInteractiveWorkflow interactiveWorkflow) :
+            base(interactiveWorkflow, RPackageCommandId.icmdNextHistoryRepl) { }
+
+        protected override void DoOperation() => Workflow.ActiveWindow.InteractiveWindow.Operations.HistoryNext();
```
Comment (AlexanderSher) [inline src/Package/Impl/Repl/Commands/History/NextHistoryReplCommand.cs]: Can you please check if it still synchronizes with RHistory window?

File: src/UnitTests/Core/Impl/Paths.cs [hunk 6]
```diff
@@ -3,22 +3,45 @@
...
 
 namespace Microsoft.UnitTests.Core {
     public static class Paths {
-        private static Lazy<string> VsRootLazy { get; } = Lazy.Create(() => (string)Registry.GetValue(Invariant($"HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\VisualStudio\\{Toolset.Version}"), "InstallDir", string.Empty));
-        private static Lazy<string> VsPrivateAssembliesLazy { get; } = Lazy.Create(() => Path.Combine(VsRootLazy.Value, @"PrivateAssemblies\"));
-        private static Lazy<string> VsCommonExtensionsLazy { get; } = Lazy.Create(() => Path.Combine(VsRootLazy.Value, @"CommonExtensions\"));
+        private static string _vsRoot;
+        public static string VsRoot {
+            get {
+                if (_vsRoot == null) {
```
Comment (AlexanderSher) [inline src/UnitTests/Core/Impl/Paths.cs]: We should either use `Lazy` or wrap it with `lock`, cause this field is accessed in tests from multiple threads.
I'd prefer `Lazy`.

File: src/Mocks/VisualStudio/project.15.0.json (hunk 4 above)
Comment (MikhailArkhipov) [inline src/Mocks/VisualStudio/project.15.0.json]: This thing is weird. I tried 1.3.0 got bunch of build issues. Looks like current (RC2) packages are bound to 1.2 so I changed all to 1.2 for consistency.

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**